File: example/pi.cpp
// Compute π to full double-double precision as 4·atan(1) and print it.
//
// Build against an installed multifloats package (see README.md):
//   cmake -S example -B build/example -DCMAKE_PREFIX_PATH=<install-prefix>
//   cmake --build build/example
#include <multifloats/float64x2.h>  // links with multifloats::multifloats
#include <iostream>

int main() {
  using namespace multifloats;
  float64x2 x(1.0);
  float64x2 pi = float64x2(4.0) * atan(x);  // π to ~106 bits
  std::cout << "pi = " << to_string(pi, 32) << '\n';
  return 0;
}

File: example/pi.f90
! Compute π to full double-double precision as 4·atan(1) and print it.
!
! Build against an installed multifloatsf package (see README.md).
program pi_example
  use multifloats                              ! generic atan + defined I/O
  use, intrinsic :: iso_fortran_env, only: dp => real64
  implicit none

  type(real64x2) :: x, pi

  x  = real64x2(1.0_dp)
  pi = atan(x) * 4                             ! π to ~106 bits
  print '(a,2es24.17)', "pi limbs = ", pi%limbs(1), pi%limbs(2)
  print *, pi                                  ! defined I/O: ~32 digits
end program pi_example
